CMakeLists.txt

```
@@ -2,6 +2,7 @@
 # just installs the headers to the install location, and optionally builds the
 # test program.
 cmake_minimum_required(VERSION 3.3.2)
+set(CMAKE_BUILD_TYPE "Release")  # ensure the tests are built with optimisation
```

This could make it really hard to debug, though, and specifying `-DCMAKE_BUILD_TYPE` in the usual way would be overridden.  Maybe we can do this instead?
```suggestion
if (NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE "Release")  # ensure the tests are built with optimisation
endif ()
```

---

Good idea.